src/API.jl

```diff
@@ -1428,8 +1439,9 @@ function precompile(ctx::Context, pkgs::Vector{PackageSpec}; internal_call::Bool
                     is_direct_dep = pkg in direct_deps
 
                     # stderr monitoring
-                    iob = Base.BufferStream()
-                    t_monitor = @async monitor_stderr(pkg, iob)
+                    stderr_pipe = Base.Pipe()
+                    stdout_pipe = Base.Pipe()
```

Comment:
```julia
                    stderr_pipe = Base.PipeEndpoint()
                    stdout_pipe = Base.PipeEndpoint()
```
If you only need to use them once, a PipeEndpoint is a bit simpler for that.

Otherwise, you must call
```julia
close(stderr_pipe.in)
```
later, once you are done with it.

But since you want to set up the async monitor call early, you are required to use the Pipe call, but you also must initialize it. This will give you much better performance IO anyways, so it is highly recommended.
```
    stderr_pipe =  Base.link_pipe!(Pipe(); reader_supports_async=true, writer_supports_async=true)
    ...
    close(stderr_pipe.in)
```